src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/MSBuildSdkResolver.cs

```diff
@@ -190,6 +194,22 @@ private sealed class CachedState
                         minimumVSDefinedSDKVersion);
                 }
 
+                string dotnetExe = Path.Combine(dotnetRoot, Constants.DotNet + Constants.ExeSuffix);
```

Comment:
The ExeSuffix does not exist in Linux/OS environments (executable is called `dotnet`)
It looks like you can get the path safely by using existing util class:
https://github.com/dotnet/sdk/blob/8bd19a2390a6bba4aa80d1ac3b6c5385527cc311/src/Resolvers/Microsoft.DotNet.NativeWrapper/EnvironmentProvider.cs#L52

Reply:
Just a note that this resolved is only used by VS, so is framework and Windows only. Concerns like getting the non-exe suffix do not apply.

Reply:
thanks!
I checked and `GetDotnetExeDirectory` returns the directory of dotnet executable. `DOTNET_HOST_PATH` should be equal to the path of the executable (it is in `dotnet build` case)

I added a new const "dotnet.exe" and check only that